src/main/java/org/truffleruby/RubyContext.java

```diff
@@ -178,8 +184,28 @@ public RubyContext(RubyLanguage language, TruffleLanguage.Env env) {
         // Create objects that need core classes
 
         Launcher.printTruffleTimeMetric("before-create-native-platform");
-        nativePlatform = NativePlatformFactory.createPlatform(this);
-        featureLoader.initialize(nativePlatform);
+
+        truffleNFIPlatform = options.NATIVE_INTERRUPT ? new TruffleNFIPlatform(this) : null;
+
+        DefaultRubiniusConfiguration.load(rubiniusConfiguration, this);
+
+        switch (Platform.OS) {
+            case LINUX:
+                LinuxRubiniusConfiguration.load(rubiniusConfiguration, this);
+                break;
+            case DARWIN:
+                DarwinRubiniusConfiguration.load(rubiniusConfiguration, this);
+                break;
+            case SOLARIS:
+                SolarisSparcV9RubiniusConfiguration.load(rubiniusConfiguration, this);
+                break;
+            default:
+                Log.LOGGER.severe("no Rubinius configuration for this platform");
+                break;
+        }
```

Comment:
Can you extra this and DefaultRubiniusConfiguration.load to a method?
This is too much logic inside the RubyContext contructor.

Reply:
Fixed.